pkg/capacity/capacity.go

```diff
@@ -17,11 +17,11 @@ type Capacity struct {
 
 // ResourceOracle is the structure responsible for capacity tracking
 type ResourceOracle struct {
-	storage pkg.StorageModule
+	storage *stubs.StorageModuleStub
```

Comment:
Why do you replace the interface with the stub type ?

Reply:
because zbus events functions unfortunately have different signature between both the module interface, and the generated stubs, unlike the other methods where they generate the same exact interface.

For example
```
// a module that has this method
func (m *Module) Stream(ctx context.Context) <- chan Value{
    
}

// will generate a stub like
func (m *Stub) Stream(ctx context.Context) (<-chan Value, error) {
}
```

Hence it's not possible to use the interface here.